libsolidity/ast/AsmJsonImporter.cpp

```diff
@@ -0,0 +1,297 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author julius <djudju@protonmail.com>
+ * @date 2019
+ * Converts an inlineAssembly AST from JSON format to AsmData
+
+ */
+
+#include <libsolidity/ast/AsmJsonImporter.h>
+#include <libsolidity/ast/ASTJsonImporter.h>
+#include <libsolidity/ast/Types.h>
+#include <libyul/AsmData.h>
+#include <libyul/AsmDataForward.h>
+#include <liblangutil/Exceptions.h>
+#include <liblangutil/Scanner.h>
+#include <vector>
+
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string.hpp>
+
+
+using namespace std;
+using namespace dev;
+using namespace yul;
+
+using SourceLocation = langutil::SourceLocation;
+
+SourceLocation const AsmJsonImporter::createSourceLocation(Json::Value const& _node)
+{
+	astAssert(!member(_node, "src").isNull(), "'src' can not be null");
+	string srcString = _node["src"].asString();
+	vector<string> pos;
+	boost::algorithm::split(pos, srcString, boost::is_any_of(":"));
+	int start = stoi(pos[0]);
+	int end = start + stoi(pos[1]);
+	std::shared_ptr<langutil::CharStream> source = make_shared<langutil::CharStream>("", m_sourceName);
+	return SourceLocation{ start, end,source};
+}
+
+template <class T>
+T AsmJsonImporter::createAsmNode(Json::Value const& _node)
+{
+	T r;
+	r.location = createSourceLocation(_node);
+	astAssert(!r.location.isEmpty() || !r.location.source, "Invalid source location in Asm AST");
+	return r;
+
+}
+
+Json::Value AsmJsonImporter::member(Json::Value const& _node, string const& _name)
+{
+	astAssert(_node.isMember(_name), "Node '" + _node["nodeType"].asString() +  " is missing field '" + _name + "'.");
```

Comment:
The `.asString` here could cause another exception haha :wink:

Reply:
Probably just leaving the assert as `_node["nodeType"]` should be fine.

Reply:
You mean using `_node["nodeType"]` instead of `_node["nodeType"].asString()` in the second parameter of the assert?

Reply:
Yeah. I think that should work?

Reply:
Unfortunately not, no `operator+` defined for those types. Right now I am trying `_node.asString()` but I guess that is just as prone to failure...